test/loadtime/cmd/report/main.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/tendermint/tendermint/store"
+	"github.com/tendermint/tendermint/test/loadtime/report"
+	dbm "github.com/tendermint/tm-db"
+)
+
+var (
+	db     = flag.String("database-type", "goleveldb", "the type of database holding the blockstore")
+	dir    = flag.String("data-dir", "~/.tendermint/data", "path to the directory containing the tendermint databases")
+	csvOut = flag.String("csv", "", "dump the extracted latencies as raw csv for use in additional tooling")
+)
+
+func main() {
+	flag.Parse()
+	if *db == "" {
+		log.Fatalf("must specify a database-type")
+	}
+	if *dir == "" {
+		log.Fatalf("must specify a data-dir")
+	}
+	d := strings.TrimPrefix(*dir, "~/")
+	if d != *dir {
+		h, err := os.UserHomeDir()
+		if err != nil {
+			panic(err)
+		}
+		d = h + "/" + d
+	}
+	_, err := os.Stat(d)
+	if err != nil {
+		panic(err)
+	}
+	dbType := dbm.BackendType(*db)
+	db, err := dbm.NewDB("blockstore", dbType, d)
+	if err != nil {
+		panic(err)
+	}
+	s := store.NewBlockStore(db)
+	defer s.Close()
+	r, err := report.GenerateFromBlockStore(s)
+	if err != nil {
+		panic(err)
+	}
+	if *csvOut != "" {
+		cf, err := os.Create(*csvOut)
+		if err != nil {
+			panic(err)
+		}
+		w := csv.NewWriter(cf)
+		err = w.WriteAll(toRecords(r.All))
+		if err != nil {
+			panic(err)
+		}
+		return
+	}
+
+	fmt.Printf(""+
+		"Total Valid Tx: %d\n"+
```

Comment:
I would also print the sample's size, i.e., now many transactions (and blocks?) were processed

Reply:
Oops! Forget this comment. Not enough coffee! Going to fix that now :-)